src/layout.rs

```diff
@@ -355,6 +359,28 @@ impl Layout {
         }
     }
 
+    /// Initialize an empty cache with a custom size. The cache is keyed on the layout and area, so
+    /// that subsequent calls with the same parameters are faster. The cache is a LruCache, and
+    /// grows until `cache_size` is reached.
+    ///
+    /// Returns true if the cell's value was set by this call.
+    /// Returns false if the cell's value was not set by this call, this means that another thread
+    /// has set this value or that the cache size is already initialized.
+    ///
+    /// Note that a custom cache size will be set only if this function:
+    /// * is called before [Layout::split()] otherwise, the cache size is
+    ///   [`Self::DEFAULT_CACHE_SIZE`].
+    /// * is called for the first time, subsequent calls do not modify the cache size.
+    pub fn init_cache(cache_size: usize) -> bool {
```

Comment:
What's the reason this is `pub`? Are we expecting users to call `init_cache`?

Reply:
Users can call `init_cache` explicitly, so we can't embed this call into `new`.
Since this cache is not bound to a  `Layout` instance but instead is shared between all the `Layout` instances, it makes sense for me to let this method `pub`.  